src/sdam/description/server.rs

```diff
@@ -190,10 +190,7 @@ impl PartialEq for ServerDescription {
 impl ServerDescription {
     pub(crate) fn new(address: ServerAddress) -> Self {
         Self {
-            address: ServerAddress::Tcp {
-                host: address.host().to_lowercase(),
```

Comment:
@abr-egn do you know the reasoning behind doing this? I'm not sure but it might affect some sort of equality check. We might want to preserve it to be safe.

Reply:
added it back.